2013-06-03  Magnus Granberg  <zorry@gentoo.org>

	* gcc/config/arm/arm.h (DRIVER_SELF_SPECS): Add ESP_DRIVER_SELF_SPEC.
	* gcc/config/arm/elf.h (SUBSUBTARGET_EXTRA_SPECS): Add ESP_EXTRA_SPECS.

--- a/gcc/config/arm/arm.h	2013-01-15 17:17:28.000000000 +0100
+++ b/gcc/config/arm/arm.h	2013-02-18 22:45:18.327284928 +0100
@@ -2326,6 +2326,11 @@ extern const char *host_detect_local_cpu
 # define MCPU_MTUNE_NATIVE_SPECS ""
 #endif
 
-#define DRIVER_SELF_SPECS MCPU_MTUNE_NATIVE_SPECS
-
+#ifdef ENABLE_ESP
+# define DRIVER_SELF_SPECS \
+  MCPU_MTUNE_NATIVE_SPECS, \
+  ESP_DRIVER_SELF_SPEC
+#else
+# define DRIVER_SELF_SPECS MCPU_MTUNE_NATIVE_SPECS
+#endif
 #endif /* ! GCC_ARM_H */
--- a/gcc/config/arm/elf.h	2013-01-10 21:38:27.000000000 +0100
+++ b/gcc/config/arm/elf.h	2013-05-19 02:15:49.595855825 +0200
@@ -49,7 +49,11 @@
 #endif
 
 #undef SUBSUBTARGET_EXTRA_SPECS
+#ifdef ENABLE_ESP
+#define SUBSUBTARGET_EXTRA_SPECS ESP_EXTRA_SPECS
+#else
 #define SUBSUBTARGET_EXTRA_SPECS
+#endif
 
 #ifndef ASM_SPEC
 #define ASM_SPEC "\
